Enhance Kubernetes SSI UST configuration documentation (#30769)
DataDog/documentation: content/en/tracing/trace_collection/automatic_instrumentation/single-step-apm/kubernetes.md

## Overview

> <div class="alert alert-warning">Single Step Instrumentation for Kubernetes is GA for Agent versions 7.64+</div>

Reviewer: What happened to the support for Agent versions 7.63 and lower? If Singe Step Instrumentation is GA for all Agent versions, than you should remove this whole div.

> In a Kubernetes environment, use Single Step Instrumentation (SSI) for APM to install the Datadog Agent and [instrument][3] instrument your applications with the Datadog APM SDKs, all together.

Reviewer: ```suggestion
In a Kubernetes environment, use Single Step Instrumentation (SSI) for APM to install the Datadog Agent and [instrument][3] your applications with the Datadog APM SDKs in one step.
```
I find the "in one step" phrasing clearer than "all together."

Follow these steps to enable Single Step Instrumentation across your entire cluster. This automatically sends traces from all applications written in supported languages.

> **Note:** To instrument only specific namespaces or pods, seeworkload targeting in[Advanced options](#advanced-options).

Reviewer 2: ```suggestion
**Note:** To instrument only specific namespaces or pods, see workload targeting in [Advanced options](#advanced-options).
```

1. In Datadog, go to the [Install the Datadog Agent on Kubernetes][11] page.
1. Follow the on-screen instructions to choose your installation method, select an API key, and set up the Operator or Helm repository.
1. In the **Configure `datadog-agent.yaml`** section, go to **Additional configuration** > **Application Observability**, and turn on **APM Instrumentation**.

   {{< img src="tracing/trace_collection/k8s-apm-instrumentation-toggle.jpg" alt="The configuration block for installing the Datadog Agent on Kubernetes through the Datadog app" style="width:100%;" >}}

1. Deploy the Agent using the generated configuration file.
1. Restart your applications.

## Configure Unified Service Tags

Unified Service Tags (USTs) apply consistent tags across traces, metrics, and logs, making it easier to navigate and correlate your observability data.

### Prerequisites

> To use the recommended ddTraceConfigs approach, ensure you have:

Reviewer: ```suggestion
To use the recommended ddTraceConfigs approach, ensure you have the following software components:
```

| Component | Minimum Version | Notes |
|-----------|-----------------|-------|
| datadog-agent | 7.66+ | Required for valueFrom support |

> | datadog-operator | 1.16.0+ | 1.13.0+ works with agent version override |

Reviewer: ```suggestion
| datadog-operator | 1.16.0+ | 1.13.0+ works with Agent version override |
```

| datadog-helm-chart | 3.120.0+ | Added valueFrom support |

### Recommended: Configure USTs with ddTraceConfigs

> With SSI, you can automatically extract UST values from pod labels and metadata without modifying individual deployments:

Reviewer 2: Add a note to say that `app-name` can be whatever label exists in your application that you want to use as the service name. And you can use multiple labels this way as well.

```yaml
datadog:
  # Make pod labels available as tags in Datadog
  kubernetesResourcesLabelsAsTags:
    pods:
      app-name: service
    deployments.apps:
      app-name: service
    replicasets.apps:
      app-name: service

  # Set environment globally for the entire cluster
  tags:
    - "env:production"

  apm:
    instrumentation:
      enabled: true
      targets:
        - name: my-services
          podSelector:

>             matchExpressions:

Reviewer 2: ```suggestion
            # matchLabels: ...
            matchExpressions:
```

            - key: app-name           # Target pods with this label
              operator: Exists
          ddTraceConfigs:
            - name: DD_SERVICE      # Extract service name from pod label
              valueFrom:
                fieldRef:
                  fieldPath: metadata.labels['app-name']
            # DD_VERSION automatically extracted from image tags
            # DD_ENV inherited from cluster-level tags above
```

> **For namespace-based environments**, create separate targets:

Reviewer 2: I would drop the namespace based environments for now, it's kind of confusing I think and the pod/deployment doesn't get the UST propagated, this only works well for service at the moment.

```yaml
datadog:
  apm:
    instrumentation:
      enabled: true
      targets:
        - name: production-services
          namespaceSelector:
            matchNames: [prod, production]
          ddTraceConfigs:
            - name: DD_ENV
              value: "production"
        - name: staging-services
          namespaceSelector:
            matchNames: [staging, stage]
          ddTraceConfigs:
            - name: DD_ENV
              value: "staging"
```

### Alternative: Configure USTs in deployment manifests

> If your setup doesn't use labels suitable for UST extraction, you can set USTs directly in your deployment manifests using environment variables. This approach requires modifying each deployment individually but offers precise control.

Reviewer: ```suggestion
If your setup doesn't use labels suitable for UST extraction, you can set USTs directly in your deployment manifests using environment variables. This approach requires modifying each deployment individually, but offers precise control.
```
